l2geth/rollup/client.go

```diff
@@ -445,6 +493,61 @@ func (c *Client) SyncStatus() (*SyncStatus, error) {
 	return status, nil
 }
 
+// GetLatestTransactionBatch will return the latest transaction batch
+func (c *Client) GetLatestTransactionBatch() (*Batch, []*types.Transaction, error) {
+	response, err := c.client.R().
+		SetResult(&TransactionBatchResponse{}).
+		Get("/batch/transaction/latest")
+
+	if err != nil {
+		return nil, nil, errors.New("Cannot get latest transaction batch")
+	}
+	txBatch, ok := response.Result().(*TransactionBatchResponse)
+	if !ok {
+		return nil, nil, fmt.Errorf("Cannot parse transaction batch response")
+	}
+	return parseTransactionBatchResponse(txBatch, c.signer)
+}
+
+// GetTransactionBatch will return the transaction batch by batch index
+func (c *Client) GetTransactionBatch(index uint64) (*Batch, []*types.Transaction, error) {
+	str := strconv.FormatUint(index, 10)
+	response, err := c.client.R().
+		SetResult(&TransactionBatchResponse{}).
+		SetPathParams(map[string]string{
+			"index": str,
+		}).
+		Get("/batch/transaction/index/{index}")
+
+	if err != nil {
+		return nil, nil, fmt.Errorf("Cannot get transaction batch %d", index)
+	}
+	txBatch, ok := response.Result().(*TransactionBatchResponse)
+	if !ok {
+		return nil, nil, fmt.Errorf("Cannot parse transaction batch response")
+	}
+	return parseTransactionBatchResponse(txBatch, c.signer)
+}
+
+// parseTransactionBatchResponse will turn a TransactionBatchResponse into a
+// Batch and its corresponding types.Transactions
+func parseTransactionBatchResponse(txBatch *TransactionBatchResponse, signer *types.OVMSigner) (*Batch, []*types.Transaction, error) {
+	if txBatch == nil {
+		return nil, nil, nil
```

Comment:
Should return an error here?

Reply:
Errors are currently not returned in the case where the element does not exist, but it is possible to do so. The consumer would have to do error comparison for the "element not found" case and then act appropriately. I opted to not do specific error checking client side and just return `nil` but now that I think about it, it feels like more idiomatic go to have a non-nil return value when the error is nil.

https://github.com/ethereum-optimism/optimism/blob/c7bc0cef232a1b3957e05671af8fca2a9ef9d892/l2geth/rollup/sync_service.go#L251-L257

Reply:
I've updated the error handling to use `errors.Is` for error comparison. This is the idiomatic way of doing things, see: https://golang.org/pkg/errors/

Below is from the docs

----

Is unwraps its first argument sequentially looking for an error that matches the second.
It reports whether it finds a match. It should be used in preference to simple equality checks:

```golang
if errors.Is(err, fs.ErrExist)
```

is preferable to
```golang
if err == fs.ErrExist
```